### PR TITLE
Feat/send voice

### PR DESCRIPTION
**Pull request by levanion**

### CODE REVIEW

File: apps/ui/src/components/AudioPlayer/AudioPlayer.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  onCloseClick,
+}: {
+  audioUrl: string
+  onCloseClick?: () => void
+}) => {
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [currentTime, setCurrentTime] = useState(0)
+  const [duration, setDuration] = useState(0)
+  const audioRef = useRef(null as any)
+  const progressBarRef = useRef(null as any)
```
Comment (okradze) [inline apps/ui/src/components/AudioPlayer/AudioPlayer.tsx:20]: fix ref types

**Review by okradze — CHANGES_REQUESTED** [review]

File: apps/ui/src/services/chat/useCreateChatMessageService.ts [hunk 2]
```diff
@@ -27,6 +28,7 @@ export const useCreateChatMessageService = () => {
           team_id: teamId,
           local_chat_message_ref_id: localChatMessageRefId,
           parent_id: parentId,
+          voice_url: voice_url,
```
[the remark this answers is not in the document]
Comment (levanion) [reply apps/ui/src/services/chat/useCreateChatMessageService.ts:31]: done

File: apps/ui/src/components/AudioRecorder/AudioRecorder.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+              type: audioBlob.type,
+              size: audioBlob.size,
+            },
+            audioBlob,
+          )
+          setRecordedVoice(uploadedFiles?.url)
+        }
+      }
+    } catch (e) {
+      console.log(e)
```
[the remark this answers is not in the document]
Comment (levanion) [reply apps/ui/src/components/AudioRecorder/AudioRecorder.tsx:59]: done

File: apps/ui/src/components/AudioPlayer/AudioPlayer.tsx (hunk 1 above)
Comment (levanion) [reply apps/ui/src/components/AudioPlayer/AudioPlayer.tsx:20]: done

File: apps/ui/src/components/AudioRecorder/AudioRecorder.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    try {
+      const stream = await navigator.mediaDevices.getUserMedia({ audio: true })
+      const recorder = new MediaRecorder(stream)
+
+      recorder.start()
+      setMediaRecorder(recorder)
+      setRecording(true)
+    } catch (err) {
+      console.error('Error accessing the microphone:', err)
```
[the remark this answers is not in the document]
Comment (levanion) [reply apps/ui/src/components/AudioRecorder/AudioRecorder.tsx:27]: done

File: apps/ui/src/components/AudioPlayer/AudioPlayer.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  border-radius: 0 100px 100px 0;
+
+  cursor: pointer;
+  padding: 4px;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  /* background: ${({ theme }) => theme.body.backgroundColorPrimary}; */
```
[the remark this answers is not in the document]
Comment (levanion) [reply apps/ui/src/components/AudioPlayer/AudioPlayer.tsx:136]: done

**Pull request merged**